### PR TITLE
virglrenderer: update to 1.0.1, remove 32bit archs

### PR DESCRIPTION
**Pull request by heljkon**
derived from https://github.com/kjliew/qemu-3dfx/tree/master/virgil3d/MINGW-packages

### CODE REVIEW

File: mingw-w64-virglrenderer/PKGBUILD [hunk 1]
```diff
@@ -35,11 +39,9 @@ prepare() {
 
   cd "${srcdir}/${_realname}-${_tag}"
   # https://gitlab.freedesktop.org/virgl/virglrenderer/-/issues/217
+  sed "s/\(error=switch\)/\1\',\'\-Wno\-unknown\-attributes\',\'\-Wno\-unused\-parameter/" -i meson.build
+  sed "s/\(fvisibility=hidden\)/\1\',\'\-mno\-ms\-bitfields/" -i meson.build
```
Comment (Biswa96) [inline mingw-w64-virglrenderer/PKGBUILD:43]: Please use patch file for these two commands.

Comment (Biswa96) [reply mingw-w64-virglrenderer/PKGBUILD]: Oh, you want to pass those compiler flags. I did not comprehend that in sed commands. You can use CFLAGS environment variables also. For example, `export CFLAGS+=" -Wno-something` (remember the space before the new flag).

File: mingw-w64-virglrenderer/meson.patch [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
++   '-Werror=switch','-Wno-unknown-attributes','-Wno-unused-parameter',
+    '-Wno-overlength-strings',
+    '-Wno-missing-field-initializers',
+ ]
+@@ -69,7 +69,7 @@
+ add_project_arguments(cc.get_supported_arguments(warnings), language : 'c')
+ 
+ flags = [
+-   '-fvisibility=hidden',
++   '-fvisibility=hidden','-mno-ms-bitfields',
```
Comment (Biswa96) [inline mingw-w64-virglrenderer/meson.patch:18]: wouldn't this break ABI compatibility? All other packages are built with that flag on.

Comment (heljkon) [reply mingw-w64-virglrenderer/meson.patch]: Let's try it without

Comment (Biswa96) [reply mingw-w64-virglrenderer/meson.patch]: It seems that someone forgot clang exists in mingw-w64 environment :facepalm: https://gitlab.freedesktop.org/virgl/virglrenderer/-/blob/1.0.1/src/mesa/util/macros.h?ref_type=tags#L206

Comment (heljkon) [reply mingw-w64-virglrenderer/meson.patch]: To summarize:

Without further action CLANG64 build fails with

```
2024-03-28T04:47:03.1069042Z FAILED: src/libvirgl.a.p/vrend_blitter.c.obj 
2024-03-28T04:47:03.1558402Z "clang" "-Isrc/libvirgl.a.p" "-Isrc" "-I../src" "-I../src/gallium/include" "-I../src/gallium/auxiliary" "-I../src/gallium/auxiliary/util" "-I." "-I.." "-I../src/venus" "-I../src/drm" "-I../src/drm/drm-uapi" "-Isrc/mesa" "-I../src/mesa" "-I../src/mesa/compat" "-I../src/mesa/pipe" "-I../src/mesa/util" "-fdiagnostics-color=always" "-DNDEBUG" "-D_FILE_OFFSET_BITS=64" "-Wall" "-Winvalid-pch" "-Wextra" "-Wpedantic" "-std=gnu11" "-O3" "-Werror=implicit-function-declaration" "-Werror=missing-prototypes" "-Werror=pedantic" "-Wmissing-prototypes" "-Werror=incompatible-pointer-types" "-Werror=int-to-pointer-cast" "-Werror=switch" "-Wno-overlength-strings" "-Wno-missing-field-initializers" "-fvisibility=hidden" "-imacros" "C:/_/B/src/virglrenderer-1.0.1/build/config.h" "-DHAVE_CONFIG_H=1" "-march=nocona" "-msahf" "-mtune=generic" "-O2" "-pipe" "-Wp,-D_FORTIFY_SOURCE=2" "-fstack-protector-strong" "-D__USE_MINGW_ANSI_STDIO=1" "-pthread" -MD -MQ src/libvirgl.a.p/vrend_blitter.c.obj -MF "src/libvirgl.a.p/vrend_blitter.c.obj.d" -o src/libvirgl.a.p/vrend_blitter.c.obj "-c" ../src/vrend_blitter.c
2024-03-28T04:47:03.1706382Z ../src/vrend_blitter.c:94:8: warning: unknown attribute 'gcc_struct' ignored [-Wunknown-attributes]
2024-03-28T04:47:03.3580927Z    94 | struct PACKED blit_prog_key {
2024-03-28T04:47:03.3694520Z       |        ^~~~~~
2024-03-28T04:47:03.3806780Z ../src/mesa/util/macros.h:207:35: note: expanded from macro 'PACKED'
2024-03-28T04:47:03.3809025Z   207 | #    define PACKED __attribute__((gcc_struct,__packed__))
2024-03-28T04:47:03.3810091Z       |                                   ^~~~~~~~~~
2024-03-28T04:47:03.3812350Z ../src/vrend_blitter.c:112:15: error: static assertion failed due to requirement 'sizeof(struct blit_prog_key) <= sizeof(unsigned long long)': struct blit_prog_key needs to fit into a hashtable64 key
2024-03-28T04:47:03.3814560Z   112 | static_assert(sizeof(struct blit_prog_key) <= sizeof(uint64_t),
2024-03-28T04:47:03.3815606Z       |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2024-03-28T04:47:03.3816831Z ../src/vrend_blitter.c:112:44: note: expression evaluates to '11 <= 8'
2024-03-28T04:47:03.3818043Z   112 | static_assert(sizeof(struct blit_prog_key) <= sizeof(uint64_t),
2024-03-28T04:47:03.3819096Z       |               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
2024-03-28T04:47:03.3819735Z 
2024-03-28T04:47:03.3820138Z 1 warning and 1 error generated.
2024-03-28T04:47:03.3820674Z 
```

Adding `-mno-ms-bitfields` to CFLAGS results in a successful build.  I'll commit this working version now while waiting for the experts response.

Comment (Biswa96) [reply mingw-w64-virglrenderer/meson.patch]: > Adding `-mno-ms-bitfields` to CFLAGS results in a successful build.

It's not a solution and breaks ABI.

Comment (heljkon) [reply mingw-w64-virglrenderer/meson.patch]: I've searched for an alternative to -mno-ms-bitfields and was lucky to find a patch.

By replacing the external patch file with my own I reduced the number of patches to the minimum.

I was able to run Qemu with GL-Support `qemu-system-x86_64 -display gtk,gl=on -device virtio-vga-gl` using both ucrt64 and clang64 binaries.

Are these changes acceptable?

Comment (heljkon) [reply mingw-w64-virglrenderer/meson.patch]: > It's not a solution and breaks ABI.

Finally it was commit after adding `-mno-ms-bitfields` by someone else: https://github.com/msys2/MINGW-packages/commit/dbeb972814610efc295a0be4a65053717a17560f

**Pull request closed without merging**